### PR TITLE
Admin dashboard + Invite participants enhancement

### LINKED ISSUE #328: Invite Participants component enhancements
- create a more complex input field component as in the screenshot below (this is part of the admin dashboard page)
- handle user input for phone numbers, emails
- add the form validation logic (use `YUP`, check in the code base for examples)
- use [formik-material](https://github.com/stackworx/formik-material-ui)
- responsive design (on mobile a single column should be enough)
- use `autocomplete` component from `material-ui`
- support different separators to understand when the user finishes to type an address (`space`, `;`, `enter`, `tab`)
- recognise the country (if prefix is type, as *+336...*) from the number typed and set the flag accordingly (or default to `France`)

Screenshot 2020-12-15 at 10 22 09

### LINKED ISSUE #341: Admin dashboard page
- Create the admin dashboard page (in wireframe it is called *P4 administration premium - remplir le form*)
- Use [formik-material](https://github.com/stackworx/formik-material-ui) to combine `Formik` and [material-ui](https://material-ui.com) components to make the form
- disable form button when submitting
- show a little loader within the button

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
Things to do later :

- remove admin route in App/router.tsx
- manage css in App/App.scss to not have conflict with MuiTheme
- Manage Login from App/index.tsx (for the moment it is displayed over other screens)

fix #328 , fix #341

### CODE REVIEW

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            },
+        })
+    )
+
+    const classes = useStyles()
+
+    return (
+        <ThemeProvider theme={theme}>
+            <CssBaseline />
+            <IonContent style={{ '--background': 'white' }}>
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/AdminDashboard.tsx:50]: to remove?

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                alert(JSON.stringify(values, null, 2));
+                }, 500);
+            }}
+        >
+
+        {({ submitForm, isSubmitting }) => (
+                <MuiPickersUtilsProvider utils={DateFnsUtils} locale={frLocale}>
+                    <Form>
+                        <Typography variant="h5" component="h1">
+                            Créer une discussion en visio
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/CreateRoomForm.tsx:173]: meesing translation ready everywhere

**Review by HugoGresse — CHANGES_REQUESTED** [review]
can be improved later: 
- common form components move out of the CreateRoomForm
- use yup to validate field
- add muiTheme

File: src/pages/AdminDashboard/AdminDashboard.tsx (hunk 1 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/AdminDashboard.tsx:50]: For the moment the mobile background is something like `blue`. That page in particular needs to have a white background, although there's surely another more appropriate way to override that

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            </Grid>
+            );
+        }catch(error){
+            return null;
+        } 
+    }
+
+    const renderTag = (participant, index, getTagProps) => {
+
+        let ChipElement;
```
Comment (ntchou) [inline src/pages/AdminDashboard/CreateRoomForm.tsx]: Suggestion. Could be treated in another pr
```
let ChipElement = ErrorChip;

if(isParticipantValid(participant)){
      ChipElement = Chip;
  }
```

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+            isValid = false;
+        }
+
+        return isValid;
+    }
+
+    const isParticipantValid = (participant) => {
+        return validateEmail(participant) || validatePhoneNumber(participant);
+    }
+    interface MyInputProps {
```
Comment (ntchou) [inline src/pages/AdminDashboard/CreateRoomForm.tsx:106]: it will be better to declare interfaces outside of component

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+                const newValue = [...value, event.target.value] as never[]
+                setValue(newValue);
+            }
+            break;
+          }
+          default:
+        }
+      };
+
+    const  validateEmail = (email) =>{
```
Comment (ntchou) [inline src/pages/AdminDashboard/CreateRoomForm.tsx]: Could be treated in an another pr
`validateEmail`, `validatePhoneNumber`, `isParticipantValid` could be extract into a lib/helper

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+        const regexp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        return regexp.test(email);
+    }
+
+    const validatePhoneNumber = (phoneNumber) =>{
+        let isValid = true;
+
+        try {
+            parsePhoneNumber(phoneNumber,'FR');
```
Comment (ntchou) [inline src/pages/AdminDashboard/CreateRoomForm.tsx]: 'FR' is set even we choose another country?

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+    interface MyInputProps {
+        onKeyDown: (event: object) => void;
+    }
+    interface MyParams extends AutocompleteRenderInputParams {
+        inputProps: MyInputProps;
+    }
+
+    const renderFlagIcon = (participant) => {
+        try {
+            const phoneNumber = parsePhoneNumber(participant,'FR');
```
Comment (ntchou) [inline src/pages/AdminDashboard/CreateRoomForm.tsx]: same as above

**Review by ntchou — COMMENTED** [review]
Good job. 
Some missing type and translations.

**Review by HugoGresse — APPROVED** [review]

File: src/pages/AdminDashboard/CreateRoomForm.tsx [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+            const FlagIcon = Flags[phoneNumber.country];
+            return ( 
+            <Grid container alignItems="center" className={classes.chipIconContainer}>
+                <FlagIcon title="United States" className={classes.flagIcon}/>
+                <Divider orientation="vertical" flexItem/>
+            </Grid>
+            );
+        }catch(error){
+            return null;
```
[the remark this answers is not in the document]
Comment (HugoIneat) [reply src/pages/AdminDashboard/CreateRoomForm.tsx]: I don't think it is useful to display or log as it is not an error case.

In fact parsePhoneNumber of libphonenumber-js throw an exception if the phone number has a wrong format.

In this case I use this information to not display a flag (return null element) in case of e-mail for exemple.

We can interpret this catch as : "Do not return a flag element if the input text is not a phone".
